src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForArrayCodeFixProvider.cs

```diff
@@ -53,7 +54,7 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
 
         foreach (var diagnostic in diagnostics.OrderByDescending(d => d.Location.SourceSpan.Start))
         {
-            var expression = diagnostic.Location.FindNode(getInnermostNodeForTie: true, cancellationToken);
+            var expression = diagnostic.AdditionalLocations[0].FindNode(getInnermostNodeForTie: true, cancellationToken);
```

Comment:
this was a bug before, but was not something you could generally notice as the location was the first token of these nodes and asking for the node would give you teh same thing.  but htis is more correct/safer, so i updated.